probe-builder/build-probe-binaries

```diff
@@ -0,0 +1,869 @@
+	# assume sysdig in the right version is already checked out in the directory
+	HAVE_SYSDIG_CHECKOUT=1
+else
+	SYSDIG_DIR=sysdig
+	HAVE_SYSDIG_CHECKOUT=
+fi
+SYSDIG_DIR=$(readlink -f $SYSDIG_DIR)
+BASEDIR=$(pwd)
+ARCH=$(uname -m)
+BUILDER_SOURCE=$(dirname $(readlink -f $0))
+
+if [ ! -d $BASEDIR/output ]; then
+	mkdir $BASEDIR/output
+fi
+
+PROBE_DEVICE_NAME=$(echo $PROBE_NAME | cut -f1 -d-)
+
+if [ $PROBE_NAME = "sysdigcloud-probe" ]; then
+	SYSDIG_TAG="agent/${PROBE_VERSION}"
+elif [ $PROBE_NAME = "sysdig-probe" ]; then
+	SYSDIG_TAG="${PROBE_VERSION}"
+else
+	SYSDIG_TAG="${PROBE_DEVICE_NAME}/${PROBE_VERSION}"
+fi
+
+# The previously released versions of sysdig, falco, and the agent
+# don't have one of the necessary changes that this script requires
+# (namely, the ability to run cmake from the sysdig directory but
+# specify PROBE_NAME, PROBE_VERSION, and PROBE_DEVICE_NAME). Rather
+# than change what was in the tag we made modified tags x.y.z-cfgprobe
+# that include the sysdig version x.y.z but also one additional commit
+# that allows making the probe name/version/etc configurable.
+
+if [ $SYSDIG_TAG = "0.24.1" ] || [ $SYSDIG_TAG = "agent/0.85.1" ] || [ $SYSDIG_TAG = "falco/0.12.1" ]; then
+	SYSDIG_TAG="${SYSDIG_TAG}-cfgprobe"
+fi
+
+function checkout_sysdig {
+	if [ -z "$HAVE_SYSDIG_CHECKOUT" ]
+	then
+		rm -rf $SYSDIG_DIR
+		mkdir -p $SYSDIG_DIR
+		git clone git@github.com:draios/sysdig.git $SYSDIG_DIR
+		(cd $SYSDIG_DIR && git checkout $SYSDIG_TAG)
+	fi
+	HAVE_SYSDIG_CHECKOUT=1
+}
+
+declare -A builders
+function build_probe {
+	# Skip Kernel 4.15.0-29 because probe does not build against it
+	if [ $KERNEL_RELEASE-$HASH = "4.15.0-29-generic-ea0aa038a6b9bdc4bb42152682bba6ce"  ]; then
+		echo "Temporarily skipping " $PROBE_NAME-$PROBE_VERSION-$ARCH-$KERNEL_RELEASE-$HASH.ko
+		return
+	fi
+
+	if [ -f $BASEDIR/output/$PROBE_NAME-$PROBE_VERSION-$ARCH-$KERNEL_RELEASE-$HASH.ko ] &&
+	   [ -f $BASEDIR/output/$PROBE_NAME-$PROBE_VERSION-$ARCH-$KERNEL_RELEASE-$HASH_ORIG.ko ]; then
+		echo "Skipping $PROBE_NAME-$PROBE_VERSION-$ARCH-$KERNEL_RELEASE-$HASH.ko (already built)"
+		return
+	fi
+
+	PROBE_ROOT=${PROBE_ROOT:-/build/probe}
+	if [ -n "${DOCKER_BUILDER:-}" ]
+	then
+		DOCKERFILE=$BUILDER_SOURCE/Dockerfile.$DOCKER_BUILDER
+		DOCKERFILE_TAG=$DOCKER_BUILDER
+
+		echo "Forced use of builder $DOCKERFILE_TAG"
+	elif [ -n "${BUILDER_DISTRO:-}" ]
+	then
+		if [ -e $KERNELDIR/include/generated/compile.h ]
+		then
+			BUILDER_GCC_VERSION=$(grep -Po '(?<=^#define LINUX_COMPILER "gcc version )[0-9.]+' $KERNELDIR/include/generated/compile.h)
+		elif [ -e $KERNELDIR/include/linux/compile.h ]
+		then
+			# RHEL6
+			BUILDER_GCC_VERSION=$(grep -Po '(?<=^#define LINUX_COMPILER "gcc version )[0-9.]+' $KERNELDIR/include/linux/compile.h)
+		else
+			# ancient Ubuntu gets and ancient compiler
+			BUILDER_GCC_VERSION=4.8.0
+		fi
+		BUILDER_GCC_MINOR_VERSION="${BUILDER_GCC_VERSION%.*}"
+		DOCKERFILE=$(ls $BUILDER_SOURCE/Dockerfile.${BUILDER_DISTRO}-gcc* | awk \
```

Comment:
Comment that we have a bunch of files for compiler versions. Explain what to do when this fails to build.